crates/metrics/src/futures/future_tracker.rs

```diff
@@ -13,11 +14,27 @@ use std::time::Instant;
 #[derive(Debug)]
 pub struct ExecutionTime<Output> {
     /// The time spent for real action of the future.
-    pub busy: Duration,
+    busy: Duration,
     /// The idle time of the future.
-    pub idle: Duration,
+    idle: Duration,
     /// The output of the future.
-    pub output: Output,
+    output: Output,
+}
+
+impl<Output> ExecutionTime<Output> {
+    /// Unwraps the future output and returns the execution report.
+    pub fn unwrap(self, metric: &FuturesMetrics) -> Output {
```

Comment:
Can we name this something else? Like `record_to_metrics` or something like that. 

The `unwrap` actually threw me off lol. I was reading it as `unwrap_or` for some `Result` and that made me confused about all the types.